tests/publiccloud/registration.pm

```diff
@@ -36,7 +36,15 @@ sub run {
 
 sub cleanup {
     my ($self) = @_;
-    $self->{instance}->upload_log('/var/log/cloudregister', log_name => $autotest::current_test->{name} . '-cloudregister.log');
+    if ($self->{result} =~ /fail$/i) {
+        my $prev_user = $self->{instance}->{username};
+        $self->{instance}->{username} = 'root';
+        $self->{instance}->upload_log('/etc/hosts', log_name => $autotest::current_test->{name} . '-hosts.txt');
```

Comment:
more correct would be just to use sudo with each command , if you are root it won't harm , if you regular user you will get all needed permissions

Reply:
Sorry, indeed that solution was unapplicable, asking for password, pubkey not ready. I changed it much now.